server/src/main/java/org/elasticsearch/action/admin/indices/settings/get/GetSettingsResponse.java

```diff
@@ -55,12 +85,22 @@ public String getSetting(String index, String setting) {
     @Override
     public void readFrom(StreamInput in) throws IOException {
         super.readFrom(in);
-        int size = in.readVInt();
-        ImmutableOpenMap.Builder<String, Settings> builder = ImmutableOpenMap.builder();
-        for (int i = 0; i < size; i++) {
-            builder.put(in.readString(), Settings.readSettingsFromStream(in));
+
+        int settingsSize = in.readVInt();
+        ImmutableOpenMap.Builder<String, Settings> settingsBuilder = ImmutableOpenMap.builder();
+        for (int i = 0; i < settingsSize; i++) {
+            settingsBuilder.put(in.readString(), Settings.readSettingsFromStream(in));
+        }
+        ImmutableOpenMap.Builder<String, Settings> defaultSettingsBuilder = ImmutableOpenMap.builder();
+
+        if (in.getVersion().onOrAfter(org.elasticsearch.Version.V_7_0_0_alpha1)) {
```

Comment:
given that we are backporting this to 6.x I think this version should be changed to 6.3 (provided that it makes it before we cut that release). You can also add a TODO and do this later, let's just make sure we don't forget :)

Reply:
The recommended strategy we use for BWC changes is to:
 - set the version to 7.0.0, get a green build including BWC
 - backport the change to 6.x and change the version to 6.3.0, get a green build including BWC
 - open a small change to master changing the version to 6.3.0, get a green building including BWC

If the version is changed to 6.3.0 now, it will not be possible to get a green build including BWC other than via silencing tests.